File: src/application/proxy/reverse/ReverseProxyContext.ts
export interface ReverseProxyContext {
  targetDomain: string;
  targetHost: string;
  localDomain: string;
  method: string;
  url: string;
  headers: Record<string, string | string[] | undefined>;
  rawBody?: Buffer;
}

export interface ReverseProxyResult {
  blocked: boolean;
  status: number;
  headers: Record<string, string | string[]>;
  body: string | Buffer;
  redirect?: string;
}

File: src-p/application/proxy/ReverseProxyContext.py
from dataclasses import dataclass, field
from typing import Dict, Optional, Union


@dataclass
class ReverseProxyContext:
    target_domain: str
    target_host: str
    local_domain: str
    method: str
    url: str
    headers: Dict[str, str]
    raw_body: Optional[bytes] = None


@dataclass
class ReverseProxyResult:
    blocked: bool
    status: int
    headers: Dict[str, Union[str, list]]
    body: Union[str, bytes]
    redirect: Optional[str] = None
